torch/jit/_recursive.py

```diff
@@ -258,51 +258,43 @@ def create_methods_from_stubs(concrete_type, stubs):
     defaults = [get_default_args(m.original_method) for m in stubs]
     concrete_type._create_methods(defs, rcbs, defaults)
 
-def create_script_module_for_tracing(nn_module, stubs):
+def create_script_module(nn_module, stubs_fn, share_types=True):
     """
-    Creates a new ScriptModule from an nn.Module, but always uses a fresh type.
-
-    NOTE: Only use this when we cannot guarantee type sharing will work
-          correctly. This only happens today for traced modules, where the same
-          module can produce different traced methods depending on the inputs.
+    Creates a new ScriptModule from an nn.Module
 
     Arguments:
         nn_module:  The original Python nn.Module that we are creating a ScriptModule for.
-        stubs:  ScriptMethodStubs to compile as part of the conversion process.
+        stubs_fn:  Lambda that takes an nn.Module and generates a list of ScriptMethodStubs to compile.
```

Comment:
nit: this lambda looks like it is called unconditionally, so why not just pass the list of stubs?

Reply:
If `nn_module` is a scriptmodule, it won't be called and we'll just use the scriptmodule as-is.